tidy3d/components/source.py

```diff
@@ -291,7 +294,83 @@ def amp_time(self, time: float) -> complex:
         return const * offset * oscillation * amp
 
 
-SourceTimeType = Union[GaussianPulse, ContinuousWave]
+class CustomSourceTime(Pulse):
+    """Custom source time dependence.
+
+    Note
```

Comment:
is it worth validating in `Simulation` to ensure the source time is not under sampled?  What I worry about is the user ignoring this and then getting a divergent simulation. is there a way to detect under sampling? Eg by checking the minimum of the `np.diff` of the time coordinates and comparing to the `Simulation.dt`?

Also what would be the downside of just using `.interp` instead of `sel`?